examples/frozen_layer.ipynb

```diff
@@ -106,18 +106,21 @@
     "\n",
     "    # Step 3\n",
     "    @eqx.filter_jit\n",
-    "    @ft.partial(eqx.filter_value_and_grad, arg=filter_spec)\n",
+    "    @ft.partial(eqx.filter_value_and_grad, arg=filter_spec, has_aux=True)\n",
```

Comment:
This is actually a nice example of the sort of silent performance error that it was possible to make before this PR.

In this example, the JIT'd region is currently only covering the computation of the gradients. But it should actually also include the `optim.update` and `eqx.apply_updates` lines as well, since these are also JAX operations. See the RNN example for the correct way to do it. Do you want to apply that fix to this example? This would mean that it isn't necessary to use `has_aux` here, since we won't need to preserve `model` to use after the JIT region.